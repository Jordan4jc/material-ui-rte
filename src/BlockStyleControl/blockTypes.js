const BLOCK_TYPES = [
  { label: "Paragraph", variant: "body1", style: "paragraph" },
  { label: "Blockquote", variant: "body1", style: "blockquote" },
  { label: "H1", variant: "h1", style: "header-one" },
  { label: "H2", variant: "h2", style: "header-two" },
  { label: "H3", variant: "h3", style: "header-three" },
  { label: "H4", variant: "h4", style: "header-four" },
  { label: "H5", variant: "h5", style: "header-five" },
  { label: "H6", variant: "h6", style: "header-six" }
]
export default BLOCK_TYPES
