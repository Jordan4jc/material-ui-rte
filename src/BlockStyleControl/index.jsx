import React from 'react'
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';

const BLOCK_TYPES = [
  {label: 'Paragraph', style: 'paragraph'},
  {label: 'H1', style: 'header-one'},
  {label: 'H2', style: 'header-two'},
  {label: 'H3', style: 'header-three'},
  {label: 'H4', style: 'header-four'},
  {label: 'H5', style: 'header-five'},
  {label: 'H6', style: 'header-six'}
];

const BlockStyleControl = (props) => {
  // const {editorState} = props;
  // const selection = editorState.getSelection();
  // const blockType = editorState
  //   .getCurrentContent()
  //   .getBlockForKey(selection.getStartKey())
  //   .getType();
  return (
    <FormControl style={{minWidth: 120}}>
      <Select
        value={props.value}
        onChange={e => props.onChange(e.target.value)}
        inputProps={{
          name: 'typography',
          id: 'typography-simple',
        }}
      >
        {BLOCK_TYPES.map(type => (
          <MenuItem key={type.style} value={type.style}>{type.label}</MenuItem>
        ))}
      </Select>
    </FormControl>
  );
};

export default BlockStyleControl