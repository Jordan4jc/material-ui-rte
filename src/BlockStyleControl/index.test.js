import React from "react"
import { shallow } from "enzyme"
import BlockStyleControl from "./index"

describe("BlockStyleControl", () => {
  it("Should render block style control", () => {
    const component = shallow(<BlockStyleControl />)

    expect(component).toMatchSnapshot()
  })
})
