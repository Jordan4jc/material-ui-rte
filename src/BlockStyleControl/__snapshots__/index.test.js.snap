// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockStyleControl Should render block style control 1`] = `ShallowWrapper {}`;
